File: TypeScript/URL/get-password-from-url.ts
import * as url from "url"

function main(): void {
    // Get the password from a given url.
    console.log(getPasswordFromURL("https://user:pass@www.example.com"))
}

main()

// Get the password from a given url.
function getPasswordFromURL(givenURL: string): string | null {
    return new URL(givenURL).password;
}

File: Python/URL/get_password_from_url.py
import urllib.parse


# Get the password from a given url.
def get_password_from_url(url):
    return urllib.parse.urlparse(url).password


def main():
    # Get the password from a given url.
    print(get_password_from_url("https://user:pass@example.com:80"))


main()
